### PR TITLE
PLAT-1275: Add semantic-release

### PR DESCRIPTION
**Pull request by homer0**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1275)

#### Description

TL;DR: I added `semantic-release` for automatic deployments through CircleCI.

The most important change here is that I removed the story for the changelog, as the `CHANGELOG.md` is automatically generated by `semantic-release` when a new version is published.

I created a Storybook addon that will generate the story by putting together a base file (`changelog.base.mdx`) with the `CHANGELOG.md` created by `semantic-release`. Since the file is created everytime Storybook builds, I added it to the `.gitignore`.

I also migrated the existing changelog to the format compatible with `semantic-release`, but I only had enough information to make the last release ~fully compatible.

### CODE REVIEW

**Review by chavlji — APPROVED** [review]
Awesome work! 🚀 🏆 

...and very nice structured pr.
